src/components/Onboarding/Screens/Apps/AgreementSettings.js

```diff
@@ -65,99 +69,144 @@ function AgreementSettings() {
+      updateField(['covenantFile', file])
     },
-    [updateField]
+    [updateField, setFormError]
   )
 
   const handleChallengePeriod = useCallback(
     value => {
       setFormError(null)
       updateField(['challengePeriod', value])
     },
-    [updateField]
+    [updateField, setFormError]
   )
 
   const handleNextClick = useCallback(() => {
-    const error = validateAgreementSettings(title, content, challengePeriod)
+    const error = validateAgreementSettings(
+      title,
+      covenantFile,
+      challengePeriod
+    )
     setFormError(error)
 
     if (!error) {
       onConfigChange('agreement', {
         title,
-        content,
+        covenantFile,
         challengePeriod,
       })
       onNext()
     }
-  }, [onConfigChange, challengePeriod, content, onNext, title])
+  }, [onConfigChange, challengePeriod, covenantFile, onNext, title])
 
   return (
     <div>
       <Header
         title="Configure Community Agreement"
         subtitle="Create the character of your DAO"
       />
-      <Field label="Title" required>
-        <TextInput value={title} onChange={handleTitleChange} autofocus wide />
-      </Field>
-      <Field label="Content/Covenant" required>
-        <TextInput
-          value={content}
-          placeholder="Define the mission, vision, and values of your community using
-          markdown format."
-          onChange={handleContentChange}
-          multiline
-          wide
-        />
-      </Field>
-      <DurationFields
-        label={
-          <Fragment>
-            Challenge Period
-            <Help hint="What is the Challenge Period?">
-              Once a proposal has been challenged, this is the amount of time
-              the proposal's creator has to either accept a settlement or raise
-              the dispute to{' '}
-              <Link href="https://1hive.gitbook.io/celeste/">Celeste</Link>.
-            </Help>
-          </Fragment>
-        }
-        duration={challengePeriod}
-        onUpdate={handleChallengePeriod}
-      />
       <div
         css={`
-          margin-top: -${2 * GU}px;
-          margin-bottom: ${3 * GU}px;
-          font-style: italic;
-          color: ${theme.contentSecondary.alpha(0.5)};
-          ${textStyle('body3')}
+          margin-bottom: ${4 * GU}px;
         `}
       >
-        We recommend sticking with the default duration.
+        <Field label="Title" required>
+          <TextInput
+            value={title}
+            onChange={handleTitleChange}
+            autofocus
+            wide
+          />
+        </Field>
+        <div
+          css={`
+            display: flex;
+          `}
+        >
+          <FileUploaderField
+            allowedMIMETypes={['text/markdown', 'text/plain']}
+            file={covenantFile}
+            onFileUpdated={handleCovenantFileChange}
+            description={
+              <>
+                Drag and drop a document here or <TextFileUploader /> to upload
```

Comment:
I am not convinced about the current format of the warning about IPFS. What do you think of handle it in a way that denotes more of a warning signal to the user?

Reply:
We can use the `Info` component to display it as a warning. The only thing I'm worry about is to overload the user with info boxes.

I'm thinking about replacing the current one with the warning:

![image](https://user-images.githubusercontent.com/33203511/127781058-8d53f0ed-8c95-43c0-9879-162f0942cbd5.png)

What do you think @0xGabi  ?